### PR TITLE
Add cocoon autosubmit configuration

### PR DESCRIPTION
**Pull request by ricardoamador**
Test pull request to add autosubmit configuration for cocoon in .github.

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: autosubmit/cocoon_autosubmit.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  - engine-flutter-autoroll
+  - dependabot[bot]
+  - dependabot
+  - DartDevtoolWorkflowBot
+approving_reviews: 2
+approval_group: flutter-hackers
+run_ci: true
+support_no_review_revert: true
+required_checkruns:
+  - 'ci.yaml validation'
```
Comment (CaseyHillers) [inline autosubmit/cocoon_autosubmit.yml:16]: This isn't true of every repo

File: autosubmit/cocoon_autosubmit.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+default_branch: main
```
[the remark this answers is not in the document]
Comment (ricardoamador) [reply autosubmit/cocoon_autosubmit.yml:1]: Correct. So I thought about what you said by not needing this however it is removes the need for us to call into github when we need the config. There are several calls that can happen before we get a pull request event so we cannot rely on pulling the information from there each time.

Including the default branch here is just a convenience for us.

File: autosubmit/cocoon_autosubmit.yml (hunk 1 above)
Comment (ricardoamador) [reply autosubmit/cocoon_autosubmit.yml:16]: Correct. My thought was that putting the files here we would have to name them for each repo. So this one is specifically for cocoon.

File: autosubmit/cocoon_autosubmit.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (CaseyHillers) [reply autosubmit/cocoon_autosubmit.yml:1]: This config is for the entire org, so wouldn't that set it to only look at main for flutter/flutter?

As an side, I wonder if we even want this logic. I'd like to enable autosubmit to support the release branches

File: autosubmit/cocoon_autosubmit.yml (hunk 1 above)
Comment (CaseyHillers) [reply autosubmit/cocoon_autosubmit.yml:16]: How is this one for cocoon?

**Pull request closed without merging**